### ISSUE TITLE
Things to add

### ISSUE THREAD

**Issue by Defragster** [TeensyUser/doc#3]
** Portable version is handy: https://forum.pjrc.com/threads/60026-Windows-10-app-vs-downloaded-Arduino-app?p=232963&viewfull=1#post232963

** note on interrupts in general and the 1062 and dsb?

** page on TyCommander

**Comment by luni64** [comment]
> page on TyCommander

It was already there, to better find it  I renamed  Build Systems to *Compiling & Uploading* 
This is also good pace to add the info about the portable version 

> note on interrupts in general and the 1062 and dsb?

Interrupt section is a good idea. Let's see where it fits. We need to take care that all information can be found easily, otherwise we'll end up in the same situation as the forum is where one often loses sight of the wood for the trees.

**Comment by luni64** [comment]
Added info to the Arduino page: https://github.com/TeensyUser/doc/wiki/Arduino-IDE
Lot of stuff to add is still open, feel free to contribute... I'm not a regular IDE user, so my experience is quite limited here.

**Comment by FrankBoesing** [comment]
maybe mention
a) the possibilty to use an external editor and 
b) "Ausfühlriche kompilierung" (Weiß nicht, wie der Punkt auf Englisch heißt)

**Comment by TeensyUser** [comment]
a) is a good idea, but again, since I hardly use the IDE, it might be better if someone else is adding this. 
b) I think you mean "File > Preferences > Show verbose output during compiling" right? I can add that

**Comment by Defragster** [comment]
Just posted a link to this : https://forum.pjrc.com/threads/67649-Non-volatile-data-for-debugging-RTWDOG-reset?p=282962&viewfull=1#post282962

That post is here: https://forum.pjrc.com/threads/67649-Non-volatile-data-for-debugging-RTWDOG-reset?p=283519&viewfull=1#post283519

On a quick glance I didn't find a 'debug' area where adding it would fit?  

Paul is doing a blog writeup for TD 1.54 - ... not sure if it will go this deep on CrashReport

**Comment by luni64** [comment]
Added a debugging section with some subsections: 

![image](https://user-images.githubusercontent.com/12611497/125353476-e698b600-e362-11eb-9262-184e8a85dbfc.png)

Feel free to add content

**Comment by Defragster** [comment]
Awesome - will give it a look

Another thing of interest is FLEXIO and see that got it's own page!

Looking over that I found this on DMA that doesn't seem to have a page?
https://forum.pjrc.com/threads/63353-Teensy-4-1-How-to-start-using-DMA?p=266991&viewfull=1#post266991

**Comment by Defragster** [comment]
Those STUB's are not present pages?

Not sure how to add any notes and get it linked in to the sidebar area?

EDIT either wants to change the footer or the sidebar?

**Comment by Defragster** [comment]
Added at least 3 intro placeholders for those pages ...

**Comment by luni64** [comment]
Good information. I fixed the code tags, no need to enclose every line in back ticks, you can use triple backticks to start and close a code block. See here https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code for details. 

If you follow the opening triple tags by c++ you get nice syntax highlighting

**Comment by Defragster** [comment]
Thanks,  I selected the code and hit the 'code' tool and it did the .code. lines as you saw it.

I knew there were better syntax - but I was just looking to drop something in there as a placeholder at least.

Web editing is a pain and everywhere is different and other than  14,370 forum.pjrc posts I don't get out much :)  Even inserting hyper links are unique ... 

I did add a couple of comments at SFun where they had bad specs for Teensy MMod.

**Comment by luni64** [comment]
> Web editing is a pain and everywhere is different and other than 14,370 forum.pjrc posts I don't get out much :) Even inserting hyper links are unique ...

Agreed, but especially the GitHub web editors are quite nice to use. Looks like the editor prefers preselecting. E.g. if you click on the code tag button you get empty single back ticks. If you preselect your code and use the code tag button it nicely includes the block in the triple ticks.  
Also simply pasting images or attachments from the clip board is a huge time saver. 

But, as  always, if one doesn't use tools regularly one tends to forget those tricks. So, for the sake of markdown training, keep the WIKI articles coming :-))

**Comment by Defragster** [comment]
I did the block select and click ... had to re-edit a few times to get it to be viewable at all - and put in the bars to get line breaks.

Going to go add a full example and see ...

**Comment by Defragster** [comment]
Updated watchdog example - select block and <> Code tool did dots all over again.

Redid with manual ```cpp edit as your examples.

**Comment by Defragster** [comment]
Added https://github.com/TeensyUser/doc/wiki/NVRAM-usage-on-T_3.x-and-T_4.x-and-MicroMod

And renamed Debugging to add NVRAM.

Has come up a couple of times on forum ... How can an active 'state/counter' be kept.

Also demo of RESTART and Bootloader entry commands.

@luni - you can re-org as appropriate, but I managed to add a page and get it linked!

**Comment by luni64** [comment]
> Added https://github.com/TeensyUser/doc/wiki/NVRAM-usage-on-T_3.x-and-T_4.x-and-MicroMod

That's very interesting, didn't know about the on-chip NVRAM.  Will give it a try this evening. 

I'm not sure if the debugging section is the right place for the page, seems only loosely coupled to debugging? What do you think about moving it to Tips & Tricks? (I can do that if you want)

**Comment by Defragster** [comment]
As noted you can re-org for the best as you know what is where.

Common use is instead of recording State/Count to EEPROM in case of restart it is written to NVRAM.

Under the covers CrashReport used the RAM2 static nature to hold the Fault info.  When that is extended to T_3.x it seems that NVRAM might be used ?

**Comment by luni64** [comment]
Moved it to the tips&tricks section: 
![image](https://user-images.githubusercontent.com/12611497/125602280-43d82297-4ed6-4c33-82cc-4a358334ea5a.png)
